### DISCUSSION TITLE
2021-08-10 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#539] [Maintainer discussions]
### Discussion topics

- **[Retrospective](https://github.com/centerofci/mathesar/discussions/539#discussioncomment-1154530)**: Summary of last week's work.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/539#discussioncomment-1154558)**: Work planned for the next week.
- **[Updates](https://github.com/centerofci/mathesar/discussions/539#discussioncomment-1154592)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/539#discussioncomment-1154606)**: Questions for the team to answer.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-08-03 until now by milestone.

### [03. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **Issues closed:**
  - [@mathemancer] #201
- **Issues started, not yet finished:**
  - [@pavish] #295

### [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **Issues closed:**
  - [@pavish] #362
  - [@pavish] #316
  - [@dhruvkb] #171
- **New issues created:**
  - #518, created by @pavish

### [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues started, not yet finished:**
  - [@pavish] #195
- **New issues created:**
  - #524, created by @pavish

### [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@eito-fis] #449
  - [@eito-fis] #397
  - [@andyharley-cci] #364
  - [@kgodey] #326
  - [@ghislaineguerin] #324
  - [@eito-fis] #203
- **Issues started, not yet finished:**
  - [@eito-fis] #452
  - [@eito-fis] #494
  - [@pegnott] #495
- **New issues created:**
  - #531, created by @kgodey

### [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@mathemancer] #383
  - [@mathemancer] #382
  - [@mathemancer] #381
  - [@mathemancer] #377
  - [@ghislaineguerin] #238
  - [@ghislaineguerin] #234
- **Issues started, not yet finished:**
  - [@ghislaineguerin] #237
  - [@ghislaineguerin] #236
  - [@ghislaineguerin] #235
  - [@powellc] #387
  - [@mathemancer] #378
  - [@mathemancer] #380
  - [@mathemancer] #379
  - [@kgodey] #367
  - [@kgodey] #368
  - [@kgodey] #369
  - [@ghislaineguerin] #240
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #239
- **New issues created:**
  - #537, created by @kgodey

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@andyharley-cci] #534
  - [@kgodey] #479
- **Issues started, not yet finished:**
  - [@andyharley-cci] #421
  - [@mr-gabe49] #483
  - [@pavish] #358
  - [@andyharley-cci] #225
  - [@Dhruvi16] #322
- **New issues created:**
  - #534, created by @andyharley-cci
  - #528, created by @dhruvkb

### [09. Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **New issues created:**
  - #538, created by @kgodey

### [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues closed:**
  - [@pavish] #312
- **Issues started, not yet finished:**
  - [@aaditkamat] #395

**Comment by kgodey** [comment]
## Active Work

This post lists active work based on the [Active](https://github.com/centerofci/mathesar/projects/1) GitHub project. It also includes a current summary of milestones involved in active work.

### [2021-08-11] [03. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **[27 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2203.%20Tables%20from%20File%20Import%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #295
- **Issues not yet started:**
  - [@pavish] #296

### [2021-08-12] [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **[13 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2204.%20Working%20with%20Schemas%22) already closed.** 
- **Issues not yet started:**
  - [@pavish] #518

### [2021-08-16] [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #195
- **Issues not yet started:**
  - [@pavish] #197
  - [@pavish] #524

### [2021-08-23] [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[14 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@eito-fis] #452
  - [@eito-fis] #494
  - [@pegnott] #495
- **Issues not yet started:**
  - [@eito-fis] #501
  - [@eito-fis] #515
- **Issues remaining in backlog:**
  - 14 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%20Working%20with%20Tables%22) issues

### [2021-08-30] [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[8 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #237
  - [@ghislaineguerin] #236
  - [@ghislaineguerin] #235
  - [@powellc] #387
  - [@mathemancer] #378
  - [@mathemancer] #380
  - [@mathemancer] #379
  - [@kgodey] #367
  - [@kgodey] #368
  - [@kgodey] #369
  - [@ghislaineguerin] #240
  - [@ghislaineguerin] #241
  - [@ghislaineguerin] #239
- **Issues not yet started:**
  - [@powellc] #388
  - [@powellc] #247
  - [@mathemancer] #434
  - [@eito-fis] #392
  - [@eito-fis] #424
  - [@eito-fis] #425
  - [@eito-fis] #426
  - [@eito-fis] #427
  - [@eito-fis] #429
  - [@rootxnet] #404
  - [@rootxnet] #405
  - [@rootxnet] #407
  - [@rootxnet] #406
  - [@rootxnet] #408
  - [@kgodey] #409
  - [@kgodey] #410
  - [@kgodey] #411
  - [@ghislaineguerin] #537
  - [@mathemancer] #417
  - [@mathemancer] #412
  - [@powellc] #384
  - [@powellc] #385
- **Issues remaining in backlog:**
  - 20 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues

### [2021-08-30] [08. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2208.%20Working%20with%20Views%22) already closed.** 
- **Issues not yet started:**
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #456
  - [@ghislaineguerin] #466
- **Issues remaining in backlog:**
  - 8 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2208.%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2208.%20Working%20with%20Views%22) issues
  - 1 other issues

### [2021-08-31] [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-08%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@andyharley-cci] #421
  - [@mr-gabe49] #483
  - [@pavish] #358
  - [@andyharley-cci] #225
  - [@Dhruvi16] #322
- **Issues not yet started:**
  - [@mathemancer] #280
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%222021-08%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%222021-08%20improvements%22) issues
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-08%20improvements%22) issues
  - 6 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%222021-08%20improvements%22) issues
  - 2 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-08%20improvements%22) issues

### [2021-09-02] [09. Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **[3 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2209.%20Data%20Modeling%22) already closed.** 
- **Issues not yet started:**
  - [@kgodey] #538
- **Issues remaining in backlog:**
  - 4 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2209.%20Data%20Modeling%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2209.%20Data%20Modeling%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2209.%20Data%20Modeling%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2209.%20Data%20Modeling%22) issues

### [2021-09-06] [10. Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2210.%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%2210.%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Updates

- Milestones [03. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1) and [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6) have switched order to account for implementation time, but both are still due this week.

- The Mathesar update presentation to the CCI research group went well. I recorded the Zoom call and I'm considering uploading the video somewhere public (like YouTube) so that it can be a resource to future contributors or other interested parties. What do you all think?
  - @aaditkamat @powellc @dhruvkb @mr-gabe49 @obulat thanks for attending!

**Reply by ghislaineguerin** [reply]
@kgodey Good idea, I think uploading it makes sense.

**Reply by mathemancer** [reply]
I think it would be great to have as well.

**Comment by kgodey** [comment]
## Questions

@mathemancer, @eito-fis, @ghislaineguerin, @pavish: Your responses are required. Others are also welcome to respond.

### Retrospective
- What went well this week?
- What could have gone better this week?
- Are there any improvements to our process you'd like to suggest?

### Active Work
- Does your workload for this week look okay?
- We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? (mainly a question for @pavish)

### Discussion
- Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?
- Anything else you'd like to discuss?

**Reply by kgodey** [reply]
> What went well this week?

I thought the presentation went well. We're also picking up speed on the data types milestone, which is massive, so it's nice to see it moving along.

I also appreciated the discussion we had about data loading in #514 

> What could have gone better this week?
> Are there any improvements to our process you'd like to suggest?

I didn't write much code this week, I hope I can spend more time on it this week. No improvements to suggest.

> Does your workload for this week look okay?

Yes. I'm going to be prioritizing finishing up mapping types, then switch to product work for the data modeling milestone, then switch back to backend work on the email type.

Also, I created an issue for the product work so that progress on it is tracked along with everything else.

> We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? 

It seems like a lot of work, even though each milestone only has 1-3 issues in it. I'll wait for @pavish to respond, but I assume we'll end up moving out one of them, at least.

> Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?

I was wondering if it might be more helpful if we broke up the issue lists under each milestone by type of work (e.g. design, frontend, etc.) and/or sorted by assignee. Some of the lists are long and it might be easier to find your work or the type of work you're interested in.

**Reply by eito-fis** [reply]
> What went well this week?

Got to merge some PRs that had been open for a while. Also feeling much more comfortable with sqlalchemy queries and postgres system tables.

> What could have gone better this week?

Feeling my inexperience with the types side of things a bit, would like to find some time to dig into how sqlalchemy and postgres handles types more thoroughly. 

> Are there any improvements to our process you'd like to suggest?

N / A

> Does your workload for this week look okay?

Yep, seems fine.

> We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? 

N / A

> Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?

The auto-generated section are awesome, no changes to suggest.

> Anything else you'd like to discuss?

N / A

**Reply by ghislaineguerin** [reply]
> What went well this week?

Reviews have been helpful and it's good to have them in parallel with new designs so that they inform current decisions.

> What could have gone better this week?

I think I took a bit longer with the last data types as we had some UI changes that had to be made across all prototypes. A consolidated prototype with components would be best but also slows down the design initially. 

> Are there any improvements to our process you'd like to suggest?

Not for now. I will try to consolidate all the work with 'Views' into a single prototype and see how that goes.

> Does your workload for this week look okay?

It does.

> We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? (mainly a question for @pavish)

No.

> Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?

Yes, I think they are useful. Not for now.

> Anything else you'd like to discuss?

Not for now.

**Reply by mathemancer** [reply]
> What went well this week?

I enjoyed knocking out a few issues.  I also think the presentation went well.  I'm pleasantly surprised that the way of implementing type support seems to be coming together so that most type implementations should be quicker and quicker moving forward.  

> What could have gone better this week?

I kept getting further away from the master branch than I'd prefer, since I was usually working on different branches that depended on each other.  I think some of the re-organization I did w.r.t. testing and type implementations should alleviate the need to make dependent branches moving forward.

> Are there any improvements to our process you'd like to suggest?

I think we already made the main improvement I wanted this week, having to do with how to deal with failing `npm audit` results.

> Does your workload for this week look okay?

Yes, with the caveat that I may be out for parental leave at some unpredictable moment soon.

> We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? 

Nothing I see.

> Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?

I can't decide whether putting the retro and active work together would be better.  I kept scrolling back and forth to connect what was TODO with what was DONE to make sure the union of those two covered everything I could think of.  I.e., it would be nice to have some overall view of an entire milestone.  I suppose I can just click away for that, though, and it might be overwhelming here.

> Anything else you'd like to discuss?

Nope!

**Reply by pavish** [reply]
> What went well this week?

I'm glad we decided on the data loading approach. The presentation went well. The structural changes for db + schema selection is done, this stabilizes the schema store.

> What could have gone better this week?

I had a number of interviews during the week. We also had the presentation meetings, and data loading discussions. This lead to less development time. It would have been better if I had factored in all of this before the start of the week, so that the milestone deadlines could have been set accordingly.

> Are there any improvements to our process you'd like to suggest?

I think we need to improve judgement of the amount of frontend work, while setting deadlines.

> Does your workload for this week look okay?
> We have three milestones due in the next week. Are there any blockers to meeting our upcoming deadlines? (mainly a question for @pavish)

File import preview page and indicators, involve a bit of thought. The actual code is less, but we need to factor in a number of scenarios.
Schema listing page interactivity has a day's work.

I am hoping to complete milestone 03. Tables from File import and 04. Working with schemas by 16th (Monday).

However, there are 3 issues in Milestone 05. Editable tables have a number of changes, each of them would take a couple days atleast (they may lead to creation of further issues which would only become evident during development). We will have to extend the deadline for this.

> Are the auto-generated active work and retrospective sections useful? Are there any format changes you'd like to suggest?

I find it useful. No suggestions.

> Anything else you'd like to discuss?

None.

**Reply by kgodey** [reply]
@mathemancer I could do a single summary that included both retrospective and active work. There's some repeated data between both. I might try that if I have some time in the next couple of weeks.
